react-app/components/FileManager/hooks.js

```diff
@@ -0,0 +1,257 @@
+            childrenIds: newChildrenIds,
+            childrenCount: newChildrenIds.length
+          };
+        }
+      });
+
+      return newFileMap;
+    });
+  }, []);
+
+  const moveFiles = useCallback((files, source, destination) => {
+    setFileMap((currentFileMap) => {
+      const newFileMap = { ...currentFileMap };
+      const moveFileIds = new Set(files.map((f) => f.id));
+
+      const newSourceChildrenIds = source.childrenIds.filter((id) => !moveFileIds.has(id));
+      newFileMap[source.id] = {
+        ...source,
+        childrenIds: newSourceChildrenIds,
+        childrenCount: newSourceChildrenIds.length
+      };
+
+      const newDestinationChildrenIds = [...destination.childrenIds, ...files.map((f) => f.id)];
+      newFileMap[destination.id] = {
+        ...destination,
+        childrenIds: newDestinationChildrenIds,
+        childrenCount: newDestinationChildrenIds.length
+      };
+
+      files.forEach((file) => {
+        newFileMap[file.id] = {
+          ...file,
+          parentId: destination.id
+        };
+      });
+
+      return newFileMap;
+    });
+  }, []);
+
+  const createFolder = useCallback((folderName) => {
+    setFileMap((currentFileMap) => {
+      const newFileMap = { ...currentFileMap };
+      const parentId = currentFolderIdRef.current;
+      const currentFolderKeys = Object.keys(newFileMap);
+      const parentName = newFileMap[currentFolderIdRef.current].name || '';
+      const isDirectoryAdded = currentFolderKeys.some(
+        (key) => newFileMap[key].name === folderName && newFileMap[key].parentId === parentId
+      );
+
+      if (isDirectoryAdded) {
+        return newFileMap;
+      }
+
+      const newFolderId = uuidv4();
+
+      const newFolderContent = {
+        id: newFolderId,
+        name: folderName,
+        isDir: true,
+        parentId: currentFolderIdRef.current,
+        path: `/${parentName}`,
+        isNew: true,
+        isSaved: false,
+        childrenIds: [],
+        childrenCount: 0,
+        children: []
+      };
+
+      newFileMap[newFolderId] = newFolderContent;
+
+      const parent = newFileMap[currentFolderIdRef.current];
+      newFileMap[currentFolderIdRef.current] = {
+        ...parent,
+        childrenIds: [...parent.childrenIds, newFolderId],
+        children: [...parent.children, newFolderContent]
+      };
+
+      return newFileMap;
+    });
+  }, []);
+
+  const createFile = useCallback((fileName, file) => {
+    setFileMap((currentFileMap) => {
+      const newFileMap = { ...currentFileMap };
+      const parentName = newFileMap[currentFolderIdRef.current].name || '';
+      const currentFolderKeys = Object.keys(newFileMap);
+      const parentId = currentFolderIdRef.current;
+
+      const isFileAdded = currentFolderKeys.some(
+        (key) => newFileMap[key].name === fileName && newFileMap[key].parentId === parentId
+      );
+
+      if (isFileAdded) {
+        return newFileMap;
+      }
+
+      const newFileId = uuidv4();
```

Comment:
As I mentioned before, the backend is generating a uuid. Can we use that one?

Reply:
We can not do that. This is generate once the file is added to filemap object for Chonky, so that way chonky identify the file there, you can generate when I pass you the data but not here